src/asm_writing/rewriter.cpp

```diff
@@ -1148,6 +1146,20 @@ void Rewriter::_setupCall(bool has_side_effects, llvm::ArrayRef<RewriterVar*> ar
 #endif
 }
 
+void Rewriter::_callOptimalEncoding(assembler::Register tmp_reg, void* func_addr) {
```

Comment:
I think this might have an issue for non-bjit rewrites -- for those we assemble into a temporary buffer, and then memcpy the data to the slot when it's done.  So in that case I think curInstPointer() will point into the temporary buffer and not the final location.  Maybe we could have a `will_be_relocated` flag or something similar?

As a random other thought, is our bjit reenttrant?  ie is it possible that while the bjit is jitting, it calls back into itself and tries to bjit that same section.